main.go

```diff
@@ -1,18 +1,52 @@
 package main
 
 import (
+	"flag"
 	"fmt"
 	"log"
+	"os"
 )
 
 const gomodName = "go.mod"
 
 func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
 
+func run() error {
 	updatedContents, err := MergeRequires(gomodName)
 	if err != nil {
-		log.Fatal(err)
+		return fmt.Errorf("failed to merge requires: %w", err)
+	}
+
+	var inplace bool
+	flag.BoolVar(&inplace, "in-place", false, "replace the contents of go.mod with the updated contents")
+	flag.BoolVar(&inplace, "i", false, "replace the contents of go.mod with the updated contents")
+	flag.Parse()
+
+	// check if we want to replace the contents of go.mod
+	if inplace {
+		return updateInplace(gomodName, updatedContents)
 	}
 
+	// print updated contents to stdout
 	fmt.Println(string(updatedContents))
+	return nil
+}
+
+func updateInplace(modLocation string, updatedContents []byte) error {
+	// get current file info
+	info, err := os.Stat(modLocation)
+	if err != nil {
+		return fmt.Errorf("failed to get file info: %w", err)
+	}
+
+	// write updated contents to go.mod
+	if err = os.WriteFile(modLocation, updatedContents, info.Mode()); err != nil {
+		return fmt.Errorf("failed to write updated go.mod: %w", err)
+	}
+
+	return nil
 }
```

Comment:
```suggestion
func writeFilePreserveMode(filepath string, contents []byte) error {
	info, err := os.Stat(filepath)
	if err != nil {
		return fmt.Errorf("failed to get file info: %w", err)
	}

	if err = os.WriteFile(filepath, contents, info.Mode()); err != nil {
		return fmt.Errorf("failed to write to %s: %w", filepath, err)
	}

	return nil
}
```

Reply:
On second thought, I'd rather this function did not exist and the logic is inlined. Not too opinionated about this, though.

Reply:
Fair enough. I did not know there were queued changes here.